docs/contributing-to-airbyte/developing-locally.md

```diff
@@ -0,0 +1,314 @@
+
+:::
+
+The following technologies are required to build Airbyte locally.
+
+1. [`Java 21`](https://jdk.java.net/archive/)
+2. `Node 20.`
+3. `Python 3.9`
+4. `Docker`
+5. `Jq`
+
+You should also follow the [Quickstart](../using-airbyte/getting-started/oss-quickstart.md) to install `abctl`. In 
+addition to `abctl` you should also install [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installing-from-release-binaries) 
+as well as [Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+
+If you are looking to build connectors, you should also follow the installation instructions for [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md)
+
+## Connector Contributions
+
+1. [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the [`airbyte`](https://github.com/airbytehq/airbyte) repository.
+2. Clone the fork on your workstation:
+
+```bash
+git clone git@github.com:{YOUR_USERNAME}/airbyte.git
+cd airbyte
+```
+
+3. Make sure that `abctl` is running correctly with the following command `abctl local status`. Verify the status check
+was successful
+
+4. Then, build the connector image:
+    - Verify the `airbyte-ci` tool is installed by running `airbyte-ci --help`, install the command with the instructions in the Prerequisites if the command is not found.
+    - Run `airbyte-ci connectors --name source-<source-name> build` will build your connector image.
+    - Once the command is done, you will find your connector image in your local docker host: `airbyte/source-<source-name>:dev`.
+
+5. You can then load the newly built image into the `abctl` instance using:
+
+```shell
+kind load docker-image airbyte/source-<source-name>:dev -n airbyte-abctl
+```
+
+:::info
+
+The above connector image is tagged with `dev`. You can change this to use another tag if you'd like.
+
+:::
+
+- In your browser, visit [http://localhost:8000/](http://localhost:8000/)
+- Log in
+- Go to `Settings` (gear icon in lower left corner)
+- Go to `Sources` or `Destinations` (depending on which connector you are testing)
+- Update the version number to use your docker image tag (default is `dev`)
+- Click `Change` to save the changes
+
+Now when you run a sync with that connector, it will use your local docker image
+
+
+### Connector Specification Caching
+
+The Airbyte Server caches connector specifications for performance reasons. If you update the specification of a 
+connector, you will need to clear this cache so the new changes are registered. To do this:
+
+- In your browser, visit [http://localhost:8000/](http://localhost:8000/)
+- Log in
+- Go to `Settings` (gear icon in lower left corner)
+- Go to `Sources` or `Destinations` (depending on which connector you are testing)
+- Update the version number to most recent release docker image tag, e.g. `1.6.4`
+- Click `Change` to save the changes
+- Set the version back to `dev`
+- Click `Change` to save the changes
```

Comment:
cc @pedroslopez - I thought we fixed this recently (https://github.com/airbytehq/airbyte-platform-internal/pull/9791)

Reply:
@evantahler we did fix the issue where the `dev` spec wasn't refetched, but as mentioned on the PR:

> This still requires users to go to the settings page and press the "update" button to re-fetch the spec (it will not dynamically update).

So the instructions here are correct in that you'll need to go to the settings page and click the update button, _however_ you shouldn't need to move to a different version and then back to dev for it to work. (the  "Update the version number to most recent release docker image tag, e.g. `1.6.4`" step shouldn't be necessary)